### ISSUE TITLE
[autoscaler] Azure SDK version needs to be pinned

### ISSUE THREAD

**Issue by GiovanniGatti** [ray-project/ray#13599]
**TLDR; We need to pin `azure-mgmt-compute==14.0.0`** 

### What is the problem?

While trying to deploy/update a ray cluster in Azure, the following error is generated:

```
$ ray up example-full.yaml 
Cluster: <hidden>

Checking Azure environment settings
2021-01-21 08:27:03,147	INFO config.py:39 -- Using subscription id: <hidden>
2021-01-21 08:27:03,147	INFO config.py:53 -- Creating/Updating Resource Group: <hidden>
Updating cluster configuration and running full setup.
Cluster Ray runtime will be restarted. Confirm [y/N]: y

<1/1> Setting up head node
  Prepared bootstrap config
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/updater.py", line 124, in run
    self.do_update()
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/updater.py", line 273, in do_update
    self.node_id, {TAG_RAY_NODE_STATUS: STATUS_WAITING_FOR_SSH})
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/azure/node_provider.py", line 29, in wrapper
    return f(self, *args, **kwargs)
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/azure/node_provider.py", line 226, in set_node_tags
    self.compute_client.virtual_machines.update(
AttributeError: 'VirtualMachinesOperations' object has no attribute 'update'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/updater.py", line 127, in run
    self.node_id, {TAG_RAY_NODE_STATUS: STATUS_UPDATE_FAILED})
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/azure/node_provider.py", line 29, in wrapper
    return f(self, *args, **kwargs)
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/azure/node_provider.py", line 226, in set_node_tags
    self.compute_client.virtual_machines.update(
AttributeError: 'VirtualMachinesOperations' object has no attribute 'update'

  Failed to setup head node.
```

*Ray version and other system information (Python version, TensorFlow version, OS):

OS: ubuntu 18.02
Python: 3.7.1
VirtualEnv: [requirements.txt](https://github.com/ray-project/ray/files/5847827/requirements.txt)

Typically,
```
$ pip install --upgrade pip
$ pip install ray azure-cli
```
[example-full.yaml](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/azure/example-full.yaml) is the same provided by ray team (except by the cluster-name/resource_group/subscription_id fields)

### Reproduction (REQUIRED)
```
$ mkvirtualenv deploy-ray
$ pip install --upgrade pip
$ pip install azure-cli ray
$ read -sp "Azure password: " AZ_PASS && echo && az login --allow-no-subscriptions -u <user_email> -p $AZ_PASS
$ ray up example-full.yaml
```

- [X ] I have verified my script runs in a clean environment and reproduces the issue.
- [ X] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by wuisawesome** [comment]
@gramhagen do you have any insight into this issue?

**Comment by gramhagen** [comment]
@GiovanniGatti what version of azure sdk (specifically azure-mgmt-compute) are you using?

**Comment by gramhagen** [comment]
sorry, seeing that now in you requirements.txt: azure-mgmt-compute==18.0.0, let me see if the sdk changed in some way that impacts this

**Comment by gramhagen** [comment]
hmm, yeah looks like the sdk changed at v 17.0.0 so some functions like `update` are prefixed to `begin_update` =/
we should probably adjust the docs to indicate there's a version limit.
@GiovanniGatti can you try running `pip install azure-mgmt-compute==14.0.0` and re-running `ray up` ?

**Comment by GiovanniGatti** [comment]
Hello, starting from a new fresh virtualenv with `$ pip install ray azure-cli azure-mgmt-compute==14.0.0` does solve this issue.

However it breaks again slightly further. Shall I create a new ticket?

```
$ ray up ~/sources/yaps/example-full.yaml --no-config-cache
Cluster: <hidden>

Checking Azure environment settings
2021-01-26 14:52:01,797 INFO config.py:39 -- Using subscription id: <hidden>
2021-01-26 14:52:01,797 INFO config.py:53 -- Creating/Updating Resource Group: <hidden>
No head node found. Launching a new cluster. Confirm [y/N]: y

Acquiring an up-to-date head node
  Launched a new head node
  Fetching the new head node
  
<1/1> Setting up head node
  Prepared bootstrap config
  New status: waiting-for-ssh
  [1/7] Waiting for SSH to become available
    Running `uptime` as a test.
    Fetched IP: <hidden>
ssh: connect to host <hidden> port 22: Connection refused
    SSH still not available (SSH command failed.), retrying in 5 seconds.
...
ssh: connect to host <hidden> port 22: Connection refused
    SSH still not available (SSH command failed.), retrying in 5 seconds.
Warning: Permanently added '<hidden>' (ECDSA) to the list of known hosts.
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

 13:56:20 up 2 min,  1 user,  load average: 2.93, 1.01, 0.37
Shared connection to <hidden> closed.
    Success.
  Updating cluster configuration. [hash=b6e4a35cf161a0aeadf43c85d921602da74217c1]
  New status: syncing-files
  [2/7] Processing file mounts
Shared connection to <hidden> closed.
Shared connection to <hidden> closed.
  [3/7] No worker file mounts to sync
  New status: setting-up
  [4/7] Running initialization commands
Warning: Permanently added '<hidden>' (ECDSA) to the list of known hosts.
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

Connection to <hidden> closed.
  [5/7] Initalizing command runner
Shared connection to <hidden> closed.
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.40/images/create?fromImage=rayproject%2Fray-ml&tag=latest-gpu: dial unix /var/run/docker.sock: connect: permission denied
Shared connection to <hidden> closed.
  New status: update-failed
  !!!
  SSH command failed.
  !!!
  
  Failed to setup head node.
```

**Comment by gramhagen** [comment]
yes, this seems like a separate issue now where docker is not starting. i will see if I can reproduce, but it would be better to track in a new issue

**Comment by GiovanniGatti** [comment]
I've created a [new issue](https://github.com/ray-project/ray/issues/13711) to continue investigations. I believe you can close this one. Thanks.

**Comment by wuisawesome** [comment]
Thanks for the investigation everyone! It sounds like the conclusion is that `azure-mgmt-compute` broke backwards compatibility and we need to pin to `14.0.0` for now and eventually upgrade.
